lib/Utils/Backends.pm

```diff
@@ -0,0 +1,62 @@
+# Copyright (C) 2018 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package Utils::Backends;
+use strict;
+use warnings;
+
+use base qw(Exporter);
+use Exporter;
+use testapi qw(:DEFAULT);
+
+use constant {
+    BACKEND => [
+        qw(
+          is_remote_backend
+          )
+    ],
+    CONSOLES => [
+        qw(
+          use_ssh_serial_console
```

Comment:
I know that you are addressing another issue, but I wanted to mention that is a hack. And then we have a lot of calls like `check_var('BACKEND', 'ipmi') ? use_ssh_serial_console : select_console 'root-console';`.
Last time I tried to fix `select_console` with callbacks, but could not finish it quick, so postponed the idea. However, I believe we should get rid of this method in all 15 files and have `select_console "root-console"` only. Same for user-console, except that we should switch user.

Reply:
@rwx788 actually that's a path I'd choose for later. A bit more care needs to be added there, But completely feasible, specially if [poo#45146](https://progress.opensuse.org/issues/45146 is addressed properly)